modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -339,6 +370,13 @@ private void handleTransactionAbortion(Set<RowId> pendingRowIds, Runnable onAppl
 
                     // TODO: https://issues.apache.org/jira/browse/IGNITE-20124 Prevent double storage updates within primary
                     if (item.isWriteIntent()) {
```

Comment:
Why do you not use the same approach that you used for `handleTransactionCommit`? I mean, read the last element instead of scanning the entire version queue.